apps/backend/test/unit/reconciliation/pos-reconciliation.spec.ts

```diff
@@ -1,58 +1,311 @@
 import { POSReconciliationService } from './../../../src/reconciliation/pos-reconciliation.service';
-import { TransactionService } from './../../../src/transaction/transaction.service';
+import { PaymentEntity } from './../../../src/transaction/entities/payment.entity';
+import { MatchStatus } from '../../../src/common/const';
+import { PaymentMethodClassification } from '../../../src/constants';
 import { POSDepositEntity } from '../../../src/deposits/entities/pos-deposit.entity';
-import { LocationEntity } from '../../../src/location/entities/master-location-data.entity';
+import { AppLogger } from '../../../src/logger/logger.service';
 import { PaymentService } from '../../../src/transaction/payment.service';
+import { locations } from '../../mocks/const/locations';
+import { MockData } from '../../mocks/mocks';
 
 describe('POSReconciliationService', () => {
   let service: POSReconciliationService;
-  const mockedRepo = {
-    findOneOrFail: jest.fn(() => Promise.resolve({}))
-  };
-  const mockedPaymentService = {
-    // mock the query method that is used
-    query: jest.fn(() => Promise.resolve({}))
-  };
-
-  const mockedTransactionService = {
-    // mock the query method that is used
-    query: jest.fn(() => Promise.resolve({}))
-  };
+  let posDepositService: DeepMocked<PosDepositService>;
+  let paymentService: DeepMocked<PaymentService>;
+  let logger: DeepMocked<AppLogger>;
+  let payments: PaymentEntity[];
+  let deposits: POSDepositEntity[];
+  let paymentExceptions: PaymentEntity[];
+  let depositExceptions: POSDepositEntity[];
+  let matchedRoundOne: { payment: PaymentEntity; deposit: POSDepositEntity }[];
+  let matchedRoundTwo: { payment: PaymentEntity; deposit: POSDepositEntity }[];
+  let matches: { payment: PaymentEntity; deposit: POSDepositEntity }[];
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
-      providers: [
-        POSReconciliationService,
-        PosDepositService,
-        LocationService,
-        Logger,
-        {
-          provide: getRepositoryToken(POSDepositEntity),
-          useValue: mockedRepo
-        },
-        {
-          provide: getRepositoryToken(LocationEntity),
-          useValue: mockedRepo
-        },
-        {
-          provide: TransactionService,
-          useValue: mockedTransactionService
-        },
-        {
-          provide: PaymentService,
-          useValue: mockedPaymentService
-        }
-      ]
-    }).compile();
+      providers: [POSReconciliationService]
+    })
+      .useMocker(createMock)
+      .compile();
+
+    const mockPOSData = new MockData(PaymentMethodClassification.POS);
+
+    payments = mockPOSData.paymentsMock;
+    /**
+     * Set the transaction time to be 20 minutes after the transaction date for one payment
+     */
+    payments[0].transaction.transaction_time = format(
+      getTime(
+        parse(
+          payments[0].transaction.transaction_date +
```

Comment:
Can we use template literals \`${}\` here?

Also, an explanation for the 20 minutes thing? Is it just for the one test?